### PR TITLE
Windows Dockerfile update

### PR DESCRIPTION
**Pull request by mc-nv**

### CODE REVIEW

File: Dockerfile.win10.min [hunk 1]
```diff
@@ -26,7 +26,7 @@
 
 # Windows min container for Triton build
 
-ARG BASE_IMAGE=mcr.microsoft.com/dotnet/framework/sdk:4.8
+ARG BASE_IMAGE=mcr.microsoft.com/windows:10.0.19042.1889
```
Comment (Tabrizian) [inline Dockerfile.win10.min:29]: Can you share more background on this change? Is there a ticket for this?

File: Dockerfile.win10.min [hunk 2]
```diff
@@ -131,4 +136,4 @@ RUN setx PATH "%PATH%;c:\TensorRT\lib;%CUDA_INSTALL_ROOT_WP%\bin"
 # that cmake can find the packages installed by vcpkg.
 ENV CMAKE_TOOLCHAIN_FILE /vcpkg/scripts/buildsystems/vcpkg.cmake
 ENV VCPKG_TARGET_TRIPLET x64-windows
-ENTRYPOINT C:\BuildTools\VC\Auxiliary\Build\vcvars64.bat &&
+ENTRYPOINT C:\BuildTools\VC\Auxiliary\Build\vcvars64.bat &&
```
Comment (Tabrizian) [inline Dockerfile.win10.min:139]: Add new line at the end of file

File: Dockerfile.win10.min (hunk 1 above)
Comment (mc-nv) [reply Dockerfile.win10.min:29]: Hi @Tabrizian , 
here is a related issue for it:
https://github.com/triton-inference-server/server/issues/4981

File: Dockerfile.win10.min (hunk 2 above)
Comment (M2Chornyi) [reply Dockerfile.win10.min]: done

File: Dockerfile.win10.min [hunk 3]
```diff
@@ -50,15 +51,15 @@ RUN pip install grpcio-tools
 # powershell for the install seems to be required to make the command
 # wait for the install to complete before continuing. To avoid failures
 # caused by VS regressions we want to stick with a working
-# compiler. Currently this is 16.8.5. This page contains download
+# compiler. Currently this is 16.11.21. This page contains download
 # links for buildtools.
 # https://docs.microsoft.com/en-us/visualstudio/releases/2019/history#release-dates-and-build-numbers
-ADD https://download.visualstudio.microsoft.com/download/pr/20130c62-1bc8-43d6-b4f0-c20bb7c79113/145a319d79a83376915d8f855605e152ef5f6fa2b2f1d2dca411fb03722eea72/vs_BuildTools.exe vs_buildtools.exe
+ADD https://download.visualstudio.microsoft.com/download/pr/8f1eb024-006a-43f6-a372-0721f71058b3/cc5cc690ac094fbfa78dfb8e40089ba52056026579e8d8dc31e95e8ea5466df5/vs_BuildTools.exe vs_buildtools.exe
```
[the remark this answers is not in the document]
Comment (mc-nv) [reply Dockerfile.win10.min:57]: Purpose of this change is the recommendation from MS to use lates package provided by.
https://learn.microsoft.com/en-us/visualstudio/releases/2019/history#release-dates-and-build-numbers

**Review by nv-kmcgill53 — APPROVED** [review]

**Pull request merged**